storage/src/test/java/org/apache/kafka/storage/internals/log/LogSegmentsTest.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.utils.Time;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.test.TestUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class LogSegmentsTest {
+    private final TopicPartition topicPartition = new TopicPartition("topic", 0);
+    private File logDir = null;
+
+    /* create a segment with the given base offset */
+    private LogSegment createSegment(Long offset) throws IOException {
+        return LogTestUtils.createSegment(offset, logDir, 10, Time.SYSTEM);
+    }
+
+    @BeforeEach
+    public void setup() {
+        logDir = TestUtils.tempDirectory();
+    }
+
+    @AfterEach
+    public void teardown() throws IOException {
+        Utils.delete(logDir);
+    }
+
+    private void assertEntry(LogSegment segment, Map.Entry<Long, LogSegment> tested) {
+        assertEquals(segment.baseOffset(), tested.getKey());
+        assertEquals(segment, tested.getValue());
+    }
+
+    @Test
+    public void testBasicOperations() throws IOException {
+        LogSegments segments = new LogSegments(topicPartition);
```

Comment:
Could you please make `LogSegments` implement `Closable` and then apply `try-resource`?